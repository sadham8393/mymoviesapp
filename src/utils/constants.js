// URl constants

export const MOVIEDB_PATH = "https://api.themoviedb.org/3/";
export const REACT_MOVIEDB_API_KEY  = "?api_key=0e89680c8b3fa0f259b53bbcc8f6091d";
export const MOVIEDB_IMAGE_APPEND_URL = "https://image.tmdb.org/t/p/w500/";
export const TRENDING_DAY_URL = "trending/all/day";


export const NO_IMAGE_URl = "https://s3-ap-southeast-1.amazonaws.com/upcode/static/default-image.jpg";

//Common Constants
export const INPUT_TYPE_TEXT = "text";
export const INPUT_TYPE_SELECT = "select";
export const NOT_AVAILABLE = "Not Available";


//Action Constants
export const FETCH_TRENDING_MOVIES = "FETCH_TRENDING_MOVIES";
export const FETCH_TV_MOVIES = "FETCH_TV_MOVIES";
export const DELETE_TV_MOVIES = "DELETE_TV_MOVIES";
export const MOVIES_FETCHED_SUCCESS = "MOVIES_FETCHED_SUCCESS";
export const MOVIES_FETCHED_ERROR = "MOVIES_FETCHED_ERROR";
export const DELETE_MOVIE_SUCCESS = "DELETE_MOVIE_SUCCESS";
export const DELETE_MOVIE_FAILURE = "DELETE_MOVIE_FAILURE";

export const MOVIES_FETCHED_NO_RESULTS = "MOVIES_FETCHED_NO_RESULTS";

export const RESET_RESPONSE = "RESET_RESPONSE";

export const NO_RESULTS_FOUND = "No results found for the given search input";

export const SERVER_UNAVAILABE = "Server unavailable. Please try after sometime"



