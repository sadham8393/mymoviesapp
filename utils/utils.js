import configureStore from 'redux-mock-store'

export const findByTestClass = (component, className) => {
    const wrapper = component.find(`${className}`);
    return wrapper;
};

export const findByComponent = (component, innerComponent) => {
    const wrapper = component.find(innerComponent);
    return wrapper;
};

export const testStore = (initialState) => {
    let store;
    const mockStore = configureStore();
    store = mockStore(initialState)
    return store;
};