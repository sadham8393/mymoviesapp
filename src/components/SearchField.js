import React from 'react';
import PropTypes from 'prop-types';

import {INPUT_TYPE_TEXT, INPUT_TYPE_SELECT} from '../utils/constants'

const SearchField = (props) =>{
    return (
        <div className ="container search-container">
            <form action = "" onSubmit = {props.handleSearch} className="row" >
                <div className="input-field col s4">
                    <input placeholder="Search Movie/Tv" className = "validate" id="searchInput" type="text" 
                            name = "searchterm" onChange = {(e) => props.handleChange(e, INPUT_TYPE_TEXT)} required/>
                    <span className="helper-text" data-error="Please enter Movie/Tv Series and Search." ></span>
                </div>
                
                <div className="input-field col s2">
                    <select className="browser-default" name = "tvMovieType" onChange = {(e) => props.handleChange(e, INPUT_TYPE_SELECT)}>
                    <option value="" disabled defaultValue>Search By</option>
                    <option value="multi">All</option>
                    <option value="movie">Movie</option>
                    <option value="tv">TV</option>
                    </select>
                </div>

                <div className="input-field col s2">
                    <button className="waves-effect waves-light btn" disabled = {props.searchBtnDisabled} onClick = {props.handleSearch}><i className="material-icons right">search</i>Search</button>
                </div>
            </form>
        </div>
    )
}

SearchField.propTypes = { 
    handleSearch: PropTypes.func,
    handleChange : PropTypes.func,
    searchBtnDisabled : PropTypes.bool

} 
export default SearchField;