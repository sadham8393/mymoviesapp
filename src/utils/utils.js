import { NOT_AVAILABLE } from './constants';

/**
 * Convert date format from "yyyy-mm-dd" to "dd/mm/yy"
 */

export function convertDate(date){
    if(date){
        return date.toString().split("-").reverse().join("/");
    }
    return NOT_AVAILABLE;
}