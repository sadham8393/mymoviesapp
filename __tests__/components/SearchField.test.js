import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { findByTestClass, findByComponent } from '../../utils/utils';
import SearchField from '../../src/components/SearchField';

describe('<SearchField /> component', () => { 
    let wrapper,
        testProps;
    let mockFunc;
    beforeEach(() => {
        mockFunc = jest.fn();
        testProps = {
            handleSearch : mockFunc,
            handleChange : mockFunc,
            searchBtnDisabled : false
        };
        wrapper = shallow(<SearchField {...testProps} />);
    });

    it('should render correctly', () => {
        const tree = renderer.create(<SearchField {...testProps} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render search-container correctly and to be 1', () => {
        const component = findByTestClass(wrapper, '.search-container');
        expect(component.length).toBe(1);
    });

    it('should render row correctly and to be 1', () => {
        const component = findByTestClass(wrapper, 'form.row');
        expect(component.length).toBe(1);
    });

    it('should render input-field correctly and to be 3', () => {
        const component = findByTestClass(wrapper, '.input-field');
        expect(component.length).toBe(3);
    });

    it('should render textfield correctly and to be 1', () => {
        const component = findByTestClass(wrapper, '.validate');
        expect(component.length).toBe(1);
    });

    it('should emit callback when textfield value is changed and to be 1', () => {
        const textfield = findByTestClass(wrapper, '.validate');
        textfield.simulate('change', { target: { value: 'New Comment' } });
        const callback = mockFunc.mock.calls.length;
        expect(callback).toBe(1);

    });

    it('should render error div correctly and to be 1', () => {
        const component = findByTestClass(wrapper, '.helper-text');
        expect(component.length).toBe(1);
    });

    it('should render select field correctly and to be 1', () => {
        const component = findByComponent(wrapper, 'select');
        expect(component.length).toBe(1);
    });

    it('should emit callback when select field value is changed and to be 1', () => {
        const selectField = findByComponent(wrapper, 'select');
        selectField.simulate('change', { target: { value: 'New Comment' } });
        const callback = mockFunc.mock.calls.length;
        expect(callback).toBe(1);
    });

    it('should render search button correctly and to be 1', () => {
        const component = findByComponent(wrapper, 'button');
        expect(component.length).toBe(1);
    });

    it('should emit callback when search button is clicked and to be 1', () => {
        const selectField = findByComponent(wrapper, 'button');
        selectField.simulate('click');
        const callback = mockFunc.mock.calls.length;
        expect(callback).toBe(1);
    });

});