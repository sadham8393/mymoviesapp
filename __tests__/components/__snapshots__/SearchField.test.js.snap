// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> component should render correctly 1`] = `
<div
  className="container search-container"
>
  <form
    action=""
    className="row"
    onSubmit={[MockFunction]}
  >
    <div
      className="input-field col s4"
    >
      <input
        className="validate"
        id="searchInput"
        name="searchterm"
        onChange={[Function]}
        placeholder="Search Movie/Tv"
        required={true}
        type="text"
      />
      <span
        className="helper-text"
        data-error="Please enter Movie/Tv Series and Search."
      />
    </div>
    <div
      className="input-field col s2"
    >
      <select
        className="browser-default"
        name="tvMovieType"
        onChange={[Function]}
      >
        <option
          defaultValue={true}
          disabled={true}
          value=""
        >
          Search By
        </option>
        <option
          value="multi"
        >
          All
        </option>
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="tv"
        >
          TV
        </option>
      </select>
    </div>
    <div
      className="input-field col s2"
    >
      <button
        className="waves-effect waves-light btn"
        disabled={false}
        onClick={[MockFunction]}
      >
        <i
          className="material-icons right"
        >
          search
        </i>
        Search
      </button>
    </div>
  </form>
</div>
`;
